app.rb

```diff
@@ -0,0 +1,256 @@
+    if choice2 == 2
+      system 'cls'
+      break
+    end
+    system 'cls'
+  end
+end
+
+def add_teacher
+  puts 'Add a teacher'.center(32)
+  print 'age: '
+  age = gets.chomp.to_i
+  print 'Specialization: '
+  specialization = gets.chomp
+  print 'Name: '
+  name = gets.chomp
+  @teachers.push(Teacher.new(age, specialization, name))
+end
+
+def add_book
+  system 'cls'
+  puts 'Add a book'.center(32)
+  print 'Title:'
+  title = gets.chomp
+  print 'Author:'
+  author = gets.chomp
+  @books.push(Book.new(title, author))
+  loop do
+    system 'cls'
+    puts 'Book added!'
+    puts '1: Add another book'
+    puts '2: Return to main menu'
+    print 'Enter your choice: '
+    choice2 = gets.chomp.to_i
+    if choice2 == 2
+      system 'cls'
+      break
+    end
+  end
+end
+
+def loop_create_rental
+  loop do
+    create_rental
+    system 'cls'
+    puts 'Rental created!'
+    puts '1: Create another rental'
+    puts '2: Return to previous menu'
+    print 'Enter your choice: '
+    choice2 = gets.chomp.to_i
+    if choice2 == 2
+      system 'cls'
+      break
+    end
+    system 'cls'
+  end
+end
+
+def create_rental
+  system 'cls'
+  puts 'Create a rental'.center(32)
+  puts 'Select a book from the following list by number'
+  count = 0
+  @books.each do |book|
+    puts "Book #{count += 1} Title: #{book.title} Author: #{book.author}"
+  end
+  print 'Enter your choice: '
+  book_choice = gets.chomp.to_i
+  list_book_person
+  print 'Enter your choice: '
+  person_choice = gets.chomp.to_i
+  puts 'Date: '
+  date = gets.chomp
+  person = if person_choice <= @students.length
+             @students[person_choice - 1]
+           else
+             @teachers[person_choice - @students.length - 1]
+           end
+  @rentals.push(Rental.new(date, person, @books[book_choice - 1]))
+end
+
+def list_book_person
+  count = 0
+  @books.each do |book|
+    puts "Book #{count += 1} Title: #{book.title} Author: #{book.author}"
+  end
+  puts 'Select a person from the following list by number'
+  count = 0
+  @students.each do |student|
+    puts "Student #{count += 1} Name: #{student.name} Age: #{student.age} "
+  end
+  @teachers.each do |teacher|
+    puts "Teacher #{count += 1} Name: #{teacher.name} Age: #{teacher.age} "
+  end
+end
+
+def list_rentals_by_person_id
+  puts 'List rentals by person  ID'.center(32)
+  puts "\n"
+  puts "Here are available persons with their IDs \n"
+  list_person
+  print 'Enter the ID of the person: '
+  person_id = gets.chomp.to_i
+  puts "\n"
+  puts "Here are the rentals of the person with ID #{person_id} \n"
+  rentals_found = false
+  @rentals.each do |rental|
+    if rental.person.id == person_id
+      puts "Book: #{rental.book.title} Author: #{rental.book.author} \n Rental Date: #{rental.date}"
+      rentals_found = true
+    end
+  end
+  puts 'NO rental records for the provided ID.' unless rentals_found
+  puts "\n"
+  print 'press r to return to the main menu... : '
```

Comment:
### Optional
- [ ] There is a bug here also, let us kindly fix it
![Screenshot from 2023-05-26 18-37-26](https://github.com/fuadmoin/school-library/assets/46388113/d269b647-6d32-48b7-983e-b56f9ab2c3e7)